### PR TITLE
Feature/parametrize level to accept pull request

### PR DESCRIPTION
**Pull request by to-more**
I want to parametrize the acceptance level of issues.

### CODE REVIEW

**Comment by johanols** [comment]
I would love to see this feature in the plugin since we would like to force all issues to be resolved before a pull request can be merged. What is the status of this patch? Seems like there was an error building it.

**Comment by henryju** [comment]
I'm sorry but we can't accept such feature mainstream. Even issues with critical/blocker severity may sometimes be considered as false positive/won't fix. SonarQube pull request analysis should be a tool for the developer, not something that can decide if the pull request can be merged with 100% accuracy. As long as there is no way to mark an issue as won't fix/false positive directly in the P/R, we can't implement any kind of "quality gate".

**Comment by henryju** [comment]
Sorry, I misread the P/R purpose, reopening.

**Comment by henryju** [comment]
https://jira.sonarsource.com/browse/SONARGITUB-33

File: src/main/java/org/sonar/plugins/github/GlobalReport.java [hunk 1]
```diff
@@ -97,20 +101,19 @@ public String getStatusDescription() {
...
 
   private void appendSummaryBySeverity(ReportBuilder builder) {
-    appendNewIssues(builder, Severity.BLOCKER);
-    appendNewIssues(builder, Severity.CRITICAL);
-    appendNewIssues(builder, Severity.MAJOR);
-    appendNewIssues(builder, Severity.MINOR);
-    appendNewIssues(builder, Severity.INFO);
+    int from = 0;
+    int to = Severity.values().length - 1;
+    IntStream.rangeClosed(from, to).forEach(j -> printNewIssuesForMarkdown(sb, Severity.values()[to - j]));
```
Comment (sams-gleb) [inline src/main/java/org/sonar/plugins/github/GlobalReport.java]: I don't see the `sb` object in this function, could you please fix that? + In general, this functional looks interesting for me, as currently we using this github-plugin with a custom patch to break the build on major issues

Comment (to-more) [reply src/main/java/org/sonar/plugins/github/GlobalReport.java]: ready!

**Comment by serkankirbas** [comment]
We are also looking forward to have this feature. @to-more Is it possible to resolve the conflict and merge? I can test it after the conflict resolution.

**Comment by henryju** [comment]
We are focusing all our effort on the built-in branch/pullrequest analysis support in SonarCloud/SonarQube 6.7 (commercial feature). The current GitHub plugin will be supported as long as we don't have at least the same level of features with the new mechanism. But it is unlikely we'll add more to it (only fixing bugs). Feel free to fork for your special needs.

**Comment by tiny-dancer** [comment]
@henryju do you have a link to the relevant built in feature docs you're referring too? nothing sticking out on a [search](https://docs.sonarqube.org/dosearchsite.action?cql=siteSearch+~+%22pull+request%22&queryString=pull+request )

is the built in feature going to have similar github integration as well?

thanks!

**Comment by henryju** [comment]
This is currently developed in the branch plugin, but we'll probably split once we start supporting other providers (like VSTS or BitBucket)

**Pull request closed without merging**